src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
         int numberOfErrors = retryCounter.getOnErrorEvents().size();
         assertEquals(retryCount, numberOfErrors);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilSuccess() {
```

Comment:
Can you please add `Single` and `Observable` to all test names appropriately? 

Idk, but current tests names are not very obvious, would be easier to see what to expect from tests if they have names like `takeUntil_should..When..` but it's personal thing, so feel free to ignore :)